dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/validation/field/validators/StandardDateValidator.java

```diff
@@ -36,23 +39,38 @@ public FieldValidationResult validateValue(String value, ValidatableField field,
                                                Map<String, ? extends List<? extends ValidatableField>> fieldIndex) {
         value = value.startsWith("-") ? value.substring(1) : value;
 
-        for (DateTimeFormatter parser : PARSERS) {
-            if (isValidDate(value, parser)) {
+        Object yearOnlyRestriction = params.get(RESTRICT_TO_YEAR_ONLY_KEY);
+        if (yearOnlyRestriction == null) {
+            yearOnlyRestriction = false;
+        }
+        boolean restrictToYearOnly = Boolean.parseBoolean(yearOnlyRestriction.toString());
+        if (restrictToYearOnly) {
+            if (isValidDate(value, YEAR_ONLY_PARSER)) {
```

Comment:
I would extract these ifs to a methods, so that the general structure would look like:
```
if (restrictToYearOnly) {
    validateOnlyYear();
} else {
    validateCompleteDate();
}
```

Reply:
Sure